### PR TITLE
add timeout to btmgmt comamnds

### PR DESCRIPTION
**Pull request by sameraugury**

### CODE REVIEW

File: src/get_connection_info_cmd.rs [hunk 1]
```diff
@@ -11,17 +13,19 @@ pub struct GetConnectionInfoCommand {
     params: Vec<u8>,
     address: Address,
     response: Vec<u8>,
+    timeout: time::Duration,
 }
 
 impl GetConnectionInfoCommand {
-    pub fn new(ctrl_index: u16, address: &Address) -> GetConnectionInfoCommand {
+    pub fn new(ctrl_index: u16, address: &Address, to: time::Duration) -> GetConnectionInfoCommand {
```
Comment (bsphere) [inline src/get_connection_info_cmd.rs]: call it `timeout`, you won't need an assignment in the struct later

File: src/get_connections_cmd.rs [hunk 2]
```diff
@@ -10,16 +11,18 @@ pub struct GetConnectionsCommand {
     param_length: u16,
     params: Vec<u8>,
     response: Vec<u8>,
+    timeout: time::Duration,
 }
 
 impl GetConnectionsCommand {
-    pub fn new(ctrl_index: u16) -> GetConnectionsCommand {
+    pub fn new(ctrl_index: u16, to: time::Duration) -> GetConnectionsCommand {
```
Comment (bsphere) [inline src/get_connections_cmd.rs]: call it `timeout`

File: src/lib.rs [hunk 3]
```diff
@@ -99,6 +100,8 @@ impl BTMgmt {
             )
         };
 
+        // store current timestamp
```
Comment (bsphere) [inline src/lib.rs]: i don't think that the comment is necessary for such a clear line :-)
maybe call it `start` instead of now?

File: src/lib.rs [hunk 4]
```diff
@@ -118,12 +121,26 @@ impl BTMgmt {
...
-                        break;
+                        return Ok(())
                     }
                 } else {
-                    return;
+                    return Err(error::Error::UnknownError);
                 }
             }
+
+            // check comamnd timeout
```
Comment (bsphere) [inline src/lib.rs]: if the comment stays, there's a typo in "comamnd"

Comment (sameraugury) [reply src/lib.rs]: done

File: src/get_connection_info_cmd.rs (hunk 1 above)
Comment (sameraugury) [reply src/get_connection_info_cmd.rs]: done

File: src/get_connections_cmd.rs (hunk 2 above)
Comment (sameraugury) [reply src/get_connections_cmd.rs]: done

File: src/lib.rs (hunk 3 above)
Comment (sameraugury) [reply src/lib.rs]: agreed and done

**Pull request merged**